tests/common.go

```diff
@@ -1970,11 +2133,171 @@ func updateNamespace(in interface{}, namespaceMapping map[string]string) error {
+		namespace := namespaceMapping[specObj.GetNamespace()]
+		specObj.SetNamespace(namespace)
+		return nil
+	} else if specObj, ok := in.(*batchv1beta1.CronJob); ok {
+		namespace := namespaceMapping[specObj.GetNamespace()]
+		specObj.SetNamespace(namespace)
+		return nil
+	} else if specObj, ok := in.(*batchv1.Job); ok {
+		namespace := namespaceMapping[specObj.GetNamespace()]
+		specObj.SetNamespace(namespace)
+		return nil
+	} else if specObj, ok := in.(*corev1.LimitRange); ok {
+		namespace := namespaceMapping[specObj.GetNamespace()]
+		specObj.SetNamespace(namespace)
+		return nil
+	} else if specObj, ok := in.(*networkingv1beta1.Ingress); ok {
+		namespace := namespaceMapping[specObj.GetNamespace()]
+		specObj.SetNamespace(namespace)
+		return nil
+	} else if specObj, ok := in.(*monitoringv1.Prometheus); ok {
+		namespace := namespaceMapping[specObj.GetNamespace()]
+		specObj.SetNamespace(namespace)
+		return nil
+	} else if specObj, ok := in.(*monitoringv1.PrometheusRule); ok {
+		namespace := namespaceMapping[specObj.GetNamespace()]
+		specObj.SetNamespace(namespace)
+		return nil
+	} else if specObj, ok := in.(*monitoringv1.ServiceMonitor); ok {
+		namespace := namespaceMapping[specObj.GetNamespace()]
+		specObj.SetNamespace(namespace)
+		return nil
+	} else if specObj, ok := in.(*corev1.Namespace); ok {
+		namespace := namespaceMapping[specObj.GetNamespace()]
+		specObj.SetNamespace(namespace)
+		return nil
+	} else if specObj, ok := in.(*apiextensionsv1beta1.CustomResourceDefinition); ok {
+		namespace := namespaceMapping[specObj.GetNamespace()]
+		specObj.SetNamespace(namespace)
+		return nil
+	} else if specObj, ok := in.(*apiextensionsv1.CustomResourceDefinition); ok {
+		namespace := namespaceMapping[specObj.GetNamespace()]
+		specObj.SetNamespace(namespace)
+		return nil
+	} else if specObj, ok := in.(*policyv1beta1.PodDisruptionBudget); ok {
+		namespace := namespaceMapping[specObj.GetNamespace()]
+		specObj.SetNamespace(namespace)
+		return nil
+	} else if specObj, ok := in.(*netv1.NetworkPolicy); ok {
+		namespace := namespaceMapping[specObj.GetNamespace()]
+		specObj.SetNamespace(namespace)
+		return nil
+	} else if specObj, ok := in.(*corev1.Endpoints); ok {
+		namespace := namespaceMapping[specObj.GetNamespace()]
+		specObj.SetNamespace(namespace)
+		return nil
+	} else if specObj, ok := in.(*storkapi.ResourceTransformation); ok {
+		namespace := namespaceMapping[specObj.GetNamespace()]
+		specObj.SetNamespace(namespace)
+		return nil
+	} else if specObj, ok := in.(*admissionregistrationv1.ValidatingWebhookConfiguration); ok {
+		for i := range specObj.Webhooks {
+			oldns := specObj.Webhooks[i].ClientConfig.Service.Namespace
+			specObj.Webhooks[i].ClientConfig.Service.Namespace = namespaceMapping[oldns]
+		}
+		return nil
 	}
 
 	return fmt.Errorf("unsupported object while setting namespace: %v", reflect.TypeOf(in))
 }
 
+func GetSpecNameKindNamepace(specObj interface{}) (string, string, string, error) {
```

Comment:
How about giving the implementation of GetSpecNameKindNamepace below a try? I think it could make the code more readable, but please keep in the mind that I have not tested it.

```
func GetSpecNameKindNamepace(specObj interface{}) (string, string, string, error) {
    supportedTypes := map[reflect.Type]bool{
        reflect.TypeOf(&appsapi.Deployment{}):  true,
        reflect.TypeOf(&appsapi.StatefulSet{}):  true,
        reflect.TypeOf(&appsapi.DaemonSet{}): true,
        .
        .
        .
    }
    specObjType := reflect.TypeOf(specObj)
    if supportedTypes[specObjType] {
        obj := reflect.ValueOf(specObj).Elem()
        name := obj.FieldByName("Name").String()
        kind := obj.FieldByName("Kind").String()
        namespace := obj.FieldByName("Namespace").String()
        return name, kind, namespace, nil
    }
    return "", "", "", fmt.Errorf("unsupported object: %v", reflect.TypeOf(specObj))
}
```

Reply:
How about adding a docstring that describes what the function does?